documentation/start-documenting.rst

```diff
@@ -64,79 +64,109 @@ Building the documentation
+before building the documentation.
+
+
+.. _building-using-make:
+.. _using-make-make-bat:
+.. _doc-build-make:
+
+Build using Make / make.bat
+---------------------------
 
-You can also use ``make help`` to see a list of targets supported by
-:command:`make`.  Note that ``make check`` is automatically run when
-you submit a :ref:`pull request <pullrequest>`, so you should make
-sure that it runs without errors.
+A Unix ``Makefile`` is provided, :cpy-file:`Doc/Makefile`,
+along with a :cpy-file:`Doc/make.bat` batch file for Windows
+that attempts to emulate it as closely as practical.
 
-**On Windows**, the :cpy-file:`Doc/make.bat` batchfile tries to emulate
-:command:`make` as closely as possible, but the venv target is not implemented,
-so you will probably want to make sure you are working in a virtual environment
-before proceeding, otherwise all dependencies will be automatically installed
-on your system.
+.. important::
 
-When ready, run the following from the root of your :ref:`repository clone
-<checkout>` to build the output as HTML::
+   The Windows ``make.bat`` batch file lacks a ``make venv`` target,
+   automatically installing any missing dependencies
+   into your current environment.
+   Make sure the environment you `created above <doc-create-venv-windows>`__
+   is `activated <venv-activate_>`__ before running ``make.bat``.
+
+To build the docs as HTML, run::
 
-   cd Doc
    make html
 
-You can also use ``make help`` to see a list of targets supported by
-:cpy-file:`Doc/make.bat`.
+.. tip:: Substitute ``htmlview`` for ``html`` to open the docs in a web browser
+         once the build completes.
+
+To check the docs for common errors with `sphinx-lint`_
+(which is run on all :ref:`pull requests <pullrequest>`), use::
+
+   make check
+
+To list other supported :program:`Make` targets for your platform, run::
+
+   make help
+
+See :cpy-file:`Doc/README.rst` for more information.
+
+
+.. _using-sphinx-build:
+.. _doc-build-sphinx:
+
+Build using Sphinx directly
+---------------------------
+
+Advanced users may want to invoke Sphinx directly,
+to pass specialized options or to handle specific use cases.
 
-See also :cpy-file:`Doc/README.rst` for more information.
+First, ensure the environment you `created above <doc-create-venv-windows>`__
+is `activated <venv-activate_>`__.
+Then, install the documentation requirements, :cpy-file:`Doc/requirements.txt`.
+Using pip::
 
-Using sphinx-build
-------------------
+   python -m pip install --upgrade -r requirements.txt
 
-Sometimes we directly want to execute the sphinx-build tool instead of through
-``make`` (although the latter is still the preferred way). In this case, you can
-use the following command line from the ``Doc`` directory (make sure to install
-Sphinx_, blurb_ and python-docs-theme_ packages from PyPI)::
+Finally, directly invoke Sphinx with::
 
-   sphinx-build -b<builder> . build/<builder>
+   python -m sphinx -b html . build/html
```

Comment:
Also `-j auto` to 'build in parallel with N processes where possible (special value "auto" will set N to cpu-count)'?

```suggestion
   python -m sphinx -b html . build/html -j auto
```

Reply:
Hmm, perhaps, but since this is primarily intended to provide a minimal working invocation specifically for advanced users who want to pass their own custom options anyway, I'm not sure whether it makes sense to add here, considering it is entirely optional, can easily be added if desired, doesn't appear to do anything on Windows (the platform most likely to use it), and may not make sense for many use cases for which the direct invocation would be used, e.g. rebuilding only a single file or running on CI.

Reply:
Shame it doesn't work on Windows.

So it will either do nothing or make things much quicker: 62s -> 41s on my machine, which I think is a good win without much complexity. Especially if this bit is intended for advanced users.

And some Linux/Mac users who could use `make` will prefer to call things directly.

Anyway, I don't mind too much, take your pick here too!

Reply:
I'm kinda torn here...it's nice that it speeds things up for some people, but adding extra optional, non-self-documenting arguments to the baseline invocation causes me a bit of concern over complicating things for users, and them retaining it in cases where it is not helpful (e.g. rebuilding only one file). Though, on the other hand, under the same assumption that users care about the arguments they are passing, they could just as easily be aware of what `-j auto` does, or be looking up the arguments anyway.

> Shame it doesn't work on Windows.

My understanding from what Adam told me is its due to lack of `fork` support, which makes spinning up new processes very expensive and rarely worth the trouble for the types of workloads Sphinx is typically run on.

> And some Linux/Mac users who could use `make` will prefer to call things directly.

FWIW, I have a working `make` installed through Rtools on Windows (and also exclusively use Git Bash), and use it myself to build the Spyder Docs I maintain (since we have a fairly complex workflow for multi-version support), but don't use it for the CPython-related projects that use it to build docs, since it assumes an in-tree venv (whereas I'm using conda environments), I need precise control over both the Python invocation and the arguments passed to Sphinx, and I prefer a consistent invocation and behavior, with no extra magic, across the many docs projects I work on.

Reply:
Does `python -m sphinx` not accept `--parallel`, as a synonym for `-j`? If not (and AFAICT it truly, in fact, does not), _that's_ probably something that should be fixed (in sphinx), and would at least help to address the "non-self-documenting" aspect.

Reply:
(I've submitted https://github.com/sphinx-doc/sphinx/pull/11169 to do just that.)